tests/keras/test_keras_model_export.py

```diff
@@ -506,3 +507,38 @@ def test_sagemaker_docker_model_scoring_with_default_conda_env(model, model_path
     deployed_model_preds = pd.DataFrame(json.loads(scoring_response.content))
 
     np.testing.assert_array_almost_equal(deployed_model_preds.values, predicted, decimal=4)
+
+
+def test_save_model_with_tf_save_format(model_path):
+    """Ensures that Keras models can be saved with SavedModel format.
+
+    Using SavedModel format (save_format="tf") requires that the file extension
+    is _not_ "h5".
+    """
+    keras_model = mock.Mock(spec=tf.keras.Model)
+    mlflow.keras.save_model(keras_model=keras_model, path=model_path, save_format="tf")
+    _, args, kwargs = keras_model.save.mock_calls[0]
+    # Ensure that save_format propagated through
+    assert kwargs["save_format"] == "tf"
+    # Ensure that the saved model does not have h5 extension
+    assert not args[0].endswith(".h5")
+
+
+@pytest.mark.large
+def test_save_and_load_model_with_tf_save_format(tf_keras_model, model_path):
```

Comment:
Could we add tests for loading back via `pyfunc` here as well?

Reply:
Great suggestion! I added some more parameters to the `test_model_save_load` test so it covers all combinations (no param, "h5", and "tf").

This uncovered a subtle bug that I've now fixed! For clarity:

Keras will compile the model if there is an optimizer on loading, unless we explicitly request it to _not_ compile. But during prediction, compilation is forced if optimizer state is saved. In the case that the optimizer state was not saved (and thus we can't compile on load), then Keras will log a warning if we set `compile=True`. As such, the fix I applied was to call load_model with `compile=True` if `SavedModel` format was used.